msg/versioned/VehicleOdometry.msg

```diff
@@ -1,34 +1,35 @@
-# Vehicle odometry data. Fits ROS REP 147 for aerial vehicles
+# Vehicle odometry data
+#
+# Fits ROS REP 147 for aerial vehicles
 
 uint32 MESSAGE_VERSION = 0
 
-uint64 timestamp		# time since system start (microseconds)
-uint64 timestamp_sample
+uint64 timestamp # [us] Time since system start
+uint64 timestamp_sample # [us] Timestamp sample
 
+uint8 pose_frame # [@enum POSE_FRAME] Position and orientation frame of reference
 uint8 POSE_FRAME_UNKNOWN = 0
-uint8 POSE_FRAME_NED     = 1 # NED earth-fixed frame
-uint8 POSE_FRAME_FRD     = 2 # FRD world-fixed frame, arbitrary heading reference
-uint8 pose_frame            # Position and orientation frame of reference
+uint8 POSE_FRAME_NED = 1 # NED earth-fixed frame. Origin is position of GC at startup.
+uint8 POSE_FRAME_FRD = 2 # FRD world-fixed frame. Arbitrary origin and heading reference. Z is down.
 
-float32[3] position         # Position in meters. Frame of reference defined by local_frame. NaN if invalid/unknown
-float32[4] q                # Quaternion rotation from FRD body frame to reference frame. First value NaN if invalid/unknown
+float32[3] position # [m] [@frame @local_frame] [@invalid NaN If invalid/unknown] Position
+float32[4] q # [@invalid NaN First value if invalid/unknown] Quaternion rotation from FRD body frame to reference frame
```

Comment:
1. Are there units for a quaternian?
2. Wasn't sure how/whether I should use frame in this case, such as:

```
float32[4] q # [@frame @VELOCITY_FRAME_BODY_FRD] [@invalid NaN First value if invalid/unknown] Quaternion rotation of reference frame
```

Reply:
> Are there units for a quaternian?

No, it's dimensionless

Reply:
> Wasn't sure how/whether I should use frame in this case

There are various definitions for quaternions, so we should say that it follows the Hamiltonian convention (w, x, y, z, right-handed, passive rotations from body to world)

<img width="605" height="264" alt="image" src="https://github.com/user-attachments/assets/2dd62d85-71af-4ba9-b729-b81945dbb28e" />
from https://www.iri.upc.edu/people/jsola/JoanSola/objectes/notes/kinematics.pdf (section 3.1, p.33-34)